akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -2285,6 +2285,281 @@ private[akka] final class StatefulMap[S, In, Out](
   override def toString = "StatefulMap"
 }
 
+/**
+ * INTERNAL API
+ */
+private[akka] final case class StatefulMapAsync[S, In, Out](parallelism: Int)(
+    create: () => Future[S],
+    //TODO (S, In) => (S, Future[Out]) seems more comfortable for user
+    f: (S, In) => Future[(S, Out)],
+    onComplete: S => Future[Option[Out]],
+    combineState: (S, S) => S)
```

Comment:
In the `statefulMap`, always choose the right side, more like `Keep.right`.

Reply:
yep, `combineState` requires user to provide a way either choosing between "previous state" and "current state" or combine them into one state.
and if `parallelism` is 1 or completely synchronous like statefulMap then this function can always be implement as just pick the "current state".
but i'm struggling to figure out how to express this constraint in API, also how to make this operator more intuitive but somehow still keep the flexibility.
may you light me a bit?